R/teal_slices.R

```diff
@@ -132,6 +132,21 @@ c.teal_slices <- function(...) {
 }
 
 
+# this function must be defined in the `teal` namesspace so that it calls `teal::teal_slices`
+#' @noRd
+#' @keywords internal
+#'
+slices_restore <- function(file) {
```

Comment:
@chlebowa should we remove from teal.slice?

Reply:
I believe `teal.slice` should contain everything that can be done with a `teal_slice` or a `teal_slices`. The fact that we need a copy in `teal` is an unfortunate consequence of the `teal` extension of the `teal_slices` class.

Reply:
ok, good to know. slices_restore will be extended by https://github.com/insightsengineering/teal.slice/pull/432 so it's good to keep that in mind, which I know you had!

Reply:
Yes, I will copy the changes as soon as they are merged into `teal.slice`.

Reply:
Can we re-export the `teal.slices::slices_restore`?  I'm sure there's a reason, wondering what it is :grin: 

For example as `{dplyr}` [does](https://github.com/tidyverse/dplyr/blob/73742713bc61d4ef079e72b806143b23cb7c4871/R/reexport-tibble.R#L1C1-L3C19)?

Reply:
`{teal.slices}`

```R
slices_restore <- function(file, teal_slices_fun = teal_slices) {
  ....
}
```

`{teal}`

```R
slices_restore <- function(file) {
  teal.slices(file, teal_slices)
}
```

Reply:
hmmm :thinking:

Reply:
As neat as it is, it does not resolve the underlying problem (which I had missed). Consider this:
There are two versions of `teal_slices` and two respective versions `slices_restore`. Start a fresh session.

A `teal_slices` object created under `teal.slice` can be restored by both `restore` funcitons.
```
x <- teal.slice::teal_slices(
  teal.slice::teal_slice("iris", "Species")
)
teal.slice::slices_store(x, "testfile.json")
y <- teal.slice::slices_restore("testfile.json")
z <- teal:::slices_restore("testfile.json")
```
However, a similar object created under `teal` can only be loaded by the function from `teal`.
```
X <- teal::teal_slices(
  teal.slice::teal_slice("iris", "Species")
)
teal.slice::slices_store(X, "testfile.json")
Y <- teal.slice::slices_restore("testfile.json")   # FAILS
Z <- teal:::slices_restore("testfile.json")
```

After some discussion with @gogonzo and @m7pr we decided to move `slices_store` and `slices_restore` to `teal`. Only then will there be no opportunity for error.

Thank you for raising this @averissimo :+1: